src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -74,11 +76,23 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.setDefaults(kafkaProperties);
 		this.kafka = new KafkaProducer<>(kafkaProperties, new StringSerializer(), new StringSerializer());
 		this.numPartitions = kafka.partitionsFor(topic).size(); //returns 1 for new topics
-	}
-
-	public int kafkaPartition(RowMap r) {
-		String db = r.getDatabase();
-		return Math.abs(db.hashCode() % numPartitions);
+		String partitionParams = this.context.getConfig().kafkaPartitionHash + "|" + this.context.getConfig().kafkaPartitionKey;
+		switch (partitionParams) {
+			case "murmur3|database": this.partitioner = new Murmur3KafkaDatabasePartitioner(this.context.getConfig().murmur3Seed);
+				break;
+			case "murmur3|table": this.partitioner = new Murmur3KafkaTablePartitioner(this.context.getConfig().murmur3Seed);
+				break;
+			case "murmur3|primary_key": this.partitioner = new Murmur3KafkaPrimaryKeyPartitioner(this.context.getConfig().murmur3Seed);
+				break;
+			case "default|primary_key": this.partitioner = new MaxwellKafkaPrimaryKeyPartitioner();
+				break;
+			case "default|table": this.partitioner = new MaxwellKafkaTablePartitioner();
+				break;
+			case "default|database":
```

Comment:
It'd probably be simpler to have a single `KafkaPartitioner` class that was instantiated with some options:

```
  this.partitioner = new MaxwellKafkaPartitioner(PARTITION_BY_DATABASE, HASH_MURMER3, murmer3SeedValue)
```

Or you could separate the responsibility into 
1. a group of classes that choose a string for partitioning
2. a group of classes that hashes that string into an integer

either way, it doesn't make sense to have a dedicated class for each string/hash function combination.